documentation/modules/exploit/windows/http/sharepoint_data_deserialization.md

```diff
@@ -0,0 +1,100 @@
+## Vulnerable Application
+A remotely exploitable vulnerability exists within SharePoint that can be leveraged by a remote authenticated attacker
+to execute code within the context of the SharePoint application service. The vulnerability is related to a failure to
+validate the source of XML input data, leading to an unsafe deserialization operation. In a default configuration, a
+Domain User account is sufficient to access SharePoint and exploit this vulnerability.
+
+This module leverages the `/_layouts/15/quicklinks.aspx?Mode=Suggestion` endpoint that was confirmed to be vulnerable by
+[Soroush Dalili](https://twitter.com/irsdl). Alternative endpoints that instantiate the
+`ContactLinksSuggestionsMicroView` type maybe used as well but are not supported by the module.
+
+### Configuring SharePoint
+Once SharePoint is installed, it needs to be configured with a site in order to be exploitable. The Central
+Administration web interface **is not vulnerable**. To configure SharePoint to be a stand alone server:
+
+1. Install Active Directory and promote the server to be a Domain Controller
+    1. Install the "Active Directory Domain Services" role
+    1. Promote the server to a Domain Controller in a new forest
+    1. Create a Domain User account for testing
+1. Install SQL Server Express
+1. Run the "SharePoint Products Configuration Wizard"
+    1. Use the SQL Server Express instances as the database server
+1. In te SharePoint "Central Administration" console web interface:
+    1. Verify that there is a web application under the "Manage web applications" page
+    1. Create a new "Site Collection" under the "Create site collections" page
+        1. Select the previously created web application
+        1. Set a Title
+        1. Use the default "Team Site" template
+        1. Set the "Primary Site Collection Administrator" to the Domain Administrator account
+
+## Verification Steps
+
+1. Install the application and ensure a page is accessible
+1. Start msfconsole
+1. Do: `use exploit/windows/http/sharepoint_data_deserialization`
+1. Set the `RHOST`, `USERNAME`, `PASSWORD` and `PAYLOAD` options
+1. Run the exploit
```

Comment:
Should add some steps that are marked as `Optional:` which explain setting the `VHOST` and `SSL` options, as well as `DOMAIN`. This is important given you supply them in your example below.